cmd/ci/main.go

```diff
@@ -0,0 +1,354 @@
+		"conformance":   {strings.Join([]string{".", "conformance"}, string(os.PathSeparator))},
+		"unit-cli": {
+			strings.Join([]string{".", "cli", "..."}, string(os.PathSeparator)),
+			strings.Join([]string{".", "cmd", "..."}, string(os.PathSeparator)),
+			strings.Join([]string{".", "api", "..."}, string(os.PathSeparator)),
+		},
+		"unit-storage": {
+			strings.Join([]string{".", "storage", "..."}, string(os.PathSeparator)),
+			strings.Join([]string{".", "extern", "..."}, string(os.PathSeparator)),
+		},
+		"unit-node": {
+			strings.Join([]string{".", "node", "..."}, string(os.PathSeparator)),
+		},
+		"unit-rest": {
+			strings.Join([]string{".", "blockstore", "..."}, string(os.PathSeparator)),
+			strings.Join([]string{".", "build", "..."}, string(os.PathSeparator)),
+			strings.Join([]string{".", "chain", "..."}, string(os.PathSeparator)),
+			strings.Join([]string{".", "conformance", "..."}, string(os.PathSeparator)),
+			strings.Join([]string{".", "gateway", "..."}, string(os.PathSeparator)),
+			strings.Join([]string{".", "journal", "..."}, string(os.PathSeparator)),
+			strings.Join([]string{".", "lib", "..."}, string(os.PathSeparator)),
+			strings.Join([]string{".", "paychmgr", "..."}, string(os.PathSeparator)),
+			strings.Join([]string{".", "tools", "..."}, string(os.PathSeparator)),
+		},
+	}
+
+	if strings.HasPrefix(name, "itest-") {
+		return []string{strings.Join([]string{".", "itests", strings.Join([]string{strings.TrimPrefix(name, "itest-"), "test.go"}, "_")}, string(os.PathSeparator))}
+	}
+
+	return namesToPackages[name]
+}
+
+func getNeedsParameters(name string) bool {
+	names := []string{
+		"conformance",
+		"itest-api",
+		"itest-direct_data_onboard_verified",
+		"itest-direct_data_onboard",
+		"itest-manual_onboarding",
+		"itest-niporep_manual",
+		"itest-net",
+		"itest-path_detach_redeclare",
+		"itest-sealing_resources",
+		"itest-sector_import_full",
+		"itest-sector_import_simple",
+		"itest-sector_pledge",
+		"itest-sector_unseal",
+		"itest-wdpost_no_miner_storage",
+		"itest-wdpost_worker_config",
+		"itest-wdpost",
+		"itest-worker",
+		"multicore-sdr",
+		"unit-cli",
+		"unit-storage",
+	}
+	return contains(names, name)
+}
+
+func getSkipConformance(name string) bool {
+	names := []string{
+		"conformance",
+	}
+	return !contains(names, name)
+}
+
+func getTestRustProofLogs(name string) bool {
+	names := []string{
+		"multicore-sdr",
+	}
+	return contains(names, name)
+}
+
+func getFormat() string {
+	return "standard-verbose"
+}
+
+func getGoTestFlags(name string) string {
+	namesToFlags := map[string]string{
+		"multicore-sdr": "-run=TestMulticoreSDR",
+		"conformance":   "-run=TestConformance",
+	}
+	if flag, ok := namesToFlags[name]; ok {
+		return flag
+	}
+	return ""
+}
+
+func getRunners(name string) [][]string {
+	if os.Getenv("GITHUB_REPOSITORY_OWNER") != "filecoin-project" {
+		return [][]string{{"ubuntu-latest"}}
+	}
+
+	namesToRunners := map[string][][]string{
```

Comment:
I am not sure how much I like this TBH. Editing yaml for this kind of data seems much more convenient than string hardcoded into `map[string][][]string`.

From a design perspective, the code here is entirely CI specific. It should live somewhere inside `.github` folder?

I do think the long term goal is to desolve this into itests themselves in form of annotations or struct tags on "Test Suit" (see testify).

Reply:
> I do think the long term goal is to desolve this into itests themselves in form of annotations or struct tags on "Test Suit" (see testify).

I'd treat this as a mid-step towards that desired goal. It's going to be much easier to parse and manipulate the data from itests here. And then, the hardcoded info inside this tool will disappear.

Reply:
Yeah, I'd love to see more of this metadata live closer to the tests themselves with annotations, but I understand that is a bigger step.  This to me has better readability than jq commands in yaml files we have now.  I don't see this making things worse and is actually improving what we have now, so 👍 from me.

The idea of moving this into the .github directory seems good/reasonable.

Reply:
The test metadata conversation also relates to https://github.com/filecoin-project/lotus/pull/11782

Reply:
I do quite like all the command line tools living in the `cmd` directory myself. It's just a personal preference, though. I can definitely move them to the `.github` folder if you prefer. Would you like me to move the `release` tool there as well as it's mainly used in the CI?

Reply:
I'm fine with `cmd` for this one since we already have `release` in there, we can debate location for both in a separate PR I think.